ports/libidn2/portfile.cmake

```
@@ -9,62 +9,64 @@ vcpkg_download_distfile(ARCHIVE
 
 vcpkg_extract_source_archive_ex(
     OUT_SOURCE_PATH SOURCE_PATH
-    ARCHIVE ${ARCHIVE}
+    ARCHIVE "${ARCHIVE}"
     REF ${IDN2_VERSION}
 )
 
 if (VCPKG_TARGET_IS_WINDOWS)
```

```suggestion
if (VCPKG_TARGET_IS_WINDOWS OR VCPKG_TARGET_IS_MINGW)
```
If using cmake to build on mingw, this condition is also required. I noticed that you add
```
{
      "name": "vcpkg-cmake",
      "host": true,
      "platform": "windows | mingw"
    }
```
in vcpkg.json file.

---

Note that the platform expression in the manifest is not consistent with the script variables:
https://github.com/microsoft/vcpkg/blob/cdd51899f6ae3736bfbe936f9e387f86d4a63a58/scripts/cmake/vcpkg_common_definitions.cmake#L52-L55

---

Verified again: Mingw uses cmake as expected. No change needed.